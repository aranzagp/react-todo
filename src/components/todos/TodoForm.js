import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";

const TodoForm = ({
  todo,
  onSave,
  onChange,
  saving = false,
  errors = {}
}) => {
  return (
    <form onSubmit={onSave}>
      <h2>{todo.id ? "Edit" : "Add"} Todo</h2>
      {errors.onSave && (
        <div className="alert alert-danger" role="alert">
          {errors.onSave}
        </div>
      )}
      
      <TextInput
        name="title"
        label="Title"
        value={todo.title}
        onChange={onChange}
        error={errors.title}
      />

      <TextInput
        name="description"
        label="Description"
        value={todo.description}
        onChange={onChange}
        error={errors.description}
      />

      <button type="submit" disabled={saving} className="btn btn-primary">
        {saving ? "Saving..." : "Save"}
      </button>
    </form>
  );
};

TodoForm.propTypes = {
  todo: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool
};

export default TodoForm;
