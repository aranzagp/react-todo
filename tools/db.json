{"todos":[{"id":1,"title":"Llevar a los perritos a pasear","description":"Recordar llevar a los 4","done":true,"slug":"llevar-a-los-perritos-a-pasear"},{"id":2,"title":"Terminar curso de Elixir","description":"Hacer el de la subscripci√≥n","done":false,"slug":"terminar-curso-de-elixir"},{"id":3,"title":"Ir al Gym","description":"Toca hacer pierna","done":true,"slug":"ir-al-gym"}]}