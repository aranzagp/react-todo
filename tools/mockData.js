const todos = [
  {
    id: 1,
    title: "Llevar a los perritos a pasear",
    description: "Recordar llevar a los 4",
    done: true,
    slug: "llevar-a-los-perritos-a-pasear"
  },
  {
    id: 2,
    title: "Terminar curso de Elixir",
    description: "Hacer el de la subscripci√≥n",
    done: false,
    slug: "terminar-curso-de-elixir"
  },
  {
    id: 3,
    title: "Ir al Gym",
    description: "Toca hacer pierna",
    done: true,
    slug: "ir-al-gym"
  }
];

const newTodo = {
  id: null,
  title: "",
  description: "",
  done: false
};

// Using CommonJS style export so we can consume via Node (without using Babel-node)
module.exports = {
  newTodo,
  todos
};
