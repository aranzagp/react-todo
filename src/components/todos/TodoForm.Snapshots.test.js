import React from "react";
import TodoForm from "./TodoForm";
import renderer from "react-test-renderer";
import { todos } from "../../../tools/mockData";

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <TodoForm
      todo={todos[0]}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />
  );

  expect(tree).toMatchSnapshot();
});

it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <TodoForm
      todo={todos[0]}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});
