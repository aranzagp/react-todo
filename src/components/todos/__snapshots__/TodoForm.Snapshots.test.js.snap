// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Todo
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Llevar a los perritos a pasear"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="description"
    >
      Description
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="description"
        onChange={[MockFunction]}
        type="text"
        value="Recordar llevar a los 4"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Todo
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="title"
        onChange={[MockFunction]}
        type="text"
        value="Llevar a los perritos a pasear"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="description"
    >
      Description
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="description"
        onChange={[MockFunction]}
        type="text"
        value="Recordar llevar a los 4"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
